script/lib/package-application.js

```diff
@@ -4,38 +4,42 @@ const assert = require('assert')
 const childProcess = require('child_process')
 const electronPackager = require('electron-packager')
 const fs = require('fs-extra')
+const hostArch = require('electron-packager/targets').hostArch
 const includePathInPackagedApp = require('./include-path-in-packaged-app')
 const getLicenseText = require('./get-license-text')
 const path = require('path')
 const spawnSync = require('./spawn-sync')
 
 const CONFIG = require('../config')
+const HOST_ARCH = hostArch()
 
 module.exports = function () {
   const appName = getAppName()
   console.log(`Running electron-packager on ${CONFIG.intermediateAppPath} with app name "${appName}"`)
   return runPackager({
-    'appBundleId': 'com.github.atom',
-    'appCopyright': `Copyright © 2014-${(new Date()).getFullYear()} GitHub, Inc. All rights reserved.`,
-    'appVersion': CONFIG.appMetadata.version,
-    'arch': process.platform === 'darwin' ? 'x64' : process.arch, // OS X is 64-bit only
-    'asar': {unpack: buildAsarUnpackGlobExpression()},
-    'buildVersion': CONFIG.appMetadata.version,
-    'download': {cache: CONFIG.electronDownloadPath},
-    'dir': CONFIG.intermediateAppPath,
-    'extendInfo': path.join(CONFIG.repositoryRootPath, 'resources', 'mac', 'atom-Info.plist'),
-    'helperBundleId': 'com.github.atom.helper',
-    'icon': getIcon(),
-    'name': appName,
-    'out': CONFIG.buildOutputPath,
-    'overwrite': true,
-    'derefSymlinks': false,
-    'platform': process.platform,
-    'electronVersion': CONFIG.appMetadata.electronVersion,
-    'win32metadata': {
-      'CompanyName': 'GitHub, Inc.',
-      'FileDescription': 'Atom',
-      'ProductName': 'Atom'
+    appBundleId: 'com.github.atom',
+    appCopyright: `Copyright © 2014-${(new Date()).getFullYear()} GitHub, Inc. All rights reserved.`,
+    appVersion: CONFIG.appMetadata.version,
+    arch: process.platform === 'darwin' ? 'x64' : HOST_ARCH, // OS X is 64-bit only
+    asar: {unpack: buildAsarUnpackGlobExpression()},
+    buildVersion: CONFIG.appMetadata.version,
+    derefSymlinks: false,
+    download: {cache: CONFIG.electronDownloadPath},
+    dir: CONFIG.intermediateAppPath,
+    electronVersion: CONFIG.appMetadata.electronVersion,
+    extendInfo: path.join(CONFIG.repositoryRootPath, 'resources', 'mac', 'atom-Info.plist'),
+    helperBundleId: 'com.github.atom.helper',
+    icon: path.join(CONFIG.repositoryRootPath, 'resources', 'app-icons', CONFIG.channel, 'atom'),
+    name: appName,
+    out: CONFIG.buildOutputPath,
+    overwrite: true,
+    platform: process.platform,
+    // Atom doesn't have devDependencies, but if prune is true, it will delete the non-standard packageDependencies.
```

Comment:
That's actually really cool! @malept does this mean that if we enable this option we can start using `devDependencies` on the top-level `package.json` without adding them to the app package? 😍 

/cc @nathansobo , @smashwilson , @jasonrudolph

Reply:
As I mentioned in the comment, `packageDependencies` is still problematic (i.e., gets pruned) if `prune: true` is set. If Packager needs to deal with this use case, we'd have to do some refactoring of how pruning works (that is, add support for custom `*Dependencies` in `package.json` to be whitelisted).

Reply:
Oh, I understood the comment incorrectly, thanks for clarifying 👍